[Blog] You want better OpenSearch documentation?

hdhalter opened the pull request:
### Description
A blog with a recap of Eric and Charlie's presentation at OpenSearch EU for the unconference.
 
### Issues Resolved
Closes https://github.com/opensearch-project/project-website/issues/2854

### Check List
- [x] Commits are signed per the DCO using --signoff


By submitting this pull request, I confirm that my contribution is made under the terms of the BSD-3-Clause License.

epugh reviewed (approved):
Thanks for the hard work!

natebower commented on _posts/2024-06-03-so-you-want-better-opensearch-documentation.markdown:
```diff
@@ -0,0 +1,60 @@
+
+The other three quadrants, however, is where we could do a better job. There is _Tutorial_  and _Explanation_ content mixed in with the _Reference_ content, but you can spend a lot of time wading through the minutiae of how things work when you are really looking for understanding of the core principles. We also have a huge amount of great content in the existing [blog](https://opensearch.org/blo ...
+
+Charlie then asked the audience what their thoughts were and we captured a lot of great insights. I then followed up with Heather Halter whose remarks are in *italics*. 
```
"questions and comments" rather than "insights" completes the logical flow from the intro into the Q&A.

natebower commented on _posts/2024-06-03-so-you-want-better-opensearch-documentation.markdown:
```diff
@@ -0,0 +1,60 @@
+
+![lightning talk](../assets/media/blog-images/2024-06-03-improvedocs/2024-06-03-improvedocs.jpg)(https://www.youtube.com/watch?v=eRLmIzevC-4)
+
+We started with the slightly tongue-in-cheek statement that the way you search Reddit is to go to Google and search “_How do I (insert any task here) +Reddit_” to get the information you want. The pattern that lots of folks use to find information about OpenSearch (unsurprisingly, as OpenSearch started as a fork of Elasticsearch) is to go to Google and search “_How do I (insert any task here) +El ...
```
```suggestion
We started with the slightly tongue-in-cheek statement that the way you search Reddit is to go to Google and search "_How do I (insert any task here) +Reddit_" to get the information you want. The pattern that lots of folks use to find information about OpenSearch (unsurprisingly, as OpenSearch started as a fork of Elasticsearch) is to go to Google and search "_How do I (insert any task here) +Elasticsearch"_. That statement was met with a few nodding heads and light chuckling.  
```

natebower commented on _posts/2024-06-03-so-you-want-better-opensearch-documentation.markdown:
```diff
@@ -0,0 +1,60 @@
+
+_While we’ve made a lot of progress in this area, particularly with implementing a redirect strategy, we are currently planning on adding canonical links to all the older pages. The goal is to get this work done by the end of Q2, which should show a big improvement in the back half of the year._
+
+**“I often find a GitHub issue that answers my question and I ⭐it.” Someone suggested that we look for comments on GitHub issues that have lots of stars and use that as input to where docs could be improved.**
```
```suggestion
**"I often find a GitHub issue that answers my question and I ⭐ it." Someone suggested that we look for comments on GitHub issues that have lots of stars and use that as an indicator of where docs could be improved.**
```

natebower commented on _posts/2024-06-03-so-you-want-better-opensearch-documentation.markdown:
```diff
@@ -0,0 +1,60 @@
+
+ _Great suggestion!_
+
+**“My 2-yr old won’t let me work at night” was one response to the question of why people who find a gap in documentation don’t pursue opening a PR to improve the docs.** 
```
```suggestion
**"My 2-year-old won't let me work at night" was one response to the question of why people who find a gap in documentation don't pursue opening a PR to improve the docs.** 
```

natebower commented on _posts/2024-06-03-so-you-want-better-opensearch-documentation.markdown:
```diff
@@ -0,0 +1,60 @@
+
+**“Older versions of OpenSearch docs often show up in Google which is often jarring. Can we make sure Google is properly indexing the latest content?“**
+
+_While we’ve made a lot of progress in this area, particularly with implementing a redirect strategy, we are currently planning on adding canonical links to all the older pages. The goal is to get this work done by the end of Q2, which should show a big improvement in the back half of the year._
```
```suggestion
_While we've made a lot of progress in this area, particularly with implementing a redirect strategy, we are currently planning on adding canonical links to all the older pages. The goal is to get this work done by the end of Q2, which should show a big improvement in the back half of the year._
```

natebower commented on _posts/2024-06-03-so-you-want-better-opensearch-documentation.markdown:
```diff
@@ -0,0 +1,60 @@
+
+**“My 2-yr old won’t let me work at night” was one response to the question of why people who find a gap in documentation don’t pursue opening a PR to improve the docs.** 
+
+_We love community contributions and have made it really easy to start. Check out the [CONTRIBUTING.md](http://contributing.md/) file and watch this [YouTube video](https://www.youtube.com/watch?v=zmVC3fQwnVI) on our documentation process and how you can contribute. You can also search through the issues labeled "[good first issue"](https://github.com/opensearch-project/documentation-website/issu ...
```
```suggestion
_We love community contributions and have made it really easy to start. Check out the [CONTRIBUTING.md](http://contributing.md/) file and watch this [YouTube video](https://www.youtube.com/watch?v=zmVC3fQwnVI) on our documentation process and how you can contribute. You can also search through the issues labeled [good first issue](https://github.com/opensearch-project/documentation-website/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22) or [help wanted](https://github.com/opensearch-project/documentation-website/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22)._
```

natebower commented on _posts/2024-06-03-so-you-want-better-opensearch-documentation.markdown:
```diff
@@ -0,0 +1,60 @@
+
+_We love community contributions and have made it really easy to start. Check out the [CONTRIBUTING.md](http://contributing.md/) file and watch this [YouTube video](https://www.youtube.com/watch?v=zmVC3fQwnVI) on our documentation process and how you can contribute. You can also search through the issues labeled "[good first issue"](https://github.com/opensearch-project/documentation-website/issu ...
+
+**Backwards compatibility in API’s came up as an issue. It would be interesting if we could better communicate API changes over time.** 
```
```suggestion
**Backwards compatibility in APIs came up as an issue. It would be interesting if we could better communicate API changes over time.** 
```

natebower commented on _posts/2024-06-03-so-you-want-better-opensearch-documentation.markdown:
```diff
@@ -0,0 +1,60 @@
+
+**Backwards compatibility in API’s came up as an issue. It would be interesting if we could better communicate API changes over time.** 
+
+_We use the "Introduced (version)" label to indicate the release in which a feature is introduced. But going backwards to identify everything is a big job. We received a suggestion to do this for field types, so I labeled it with “*[*_good first issue_*](https://github.com/opensearch-project/documentation-website/issues/6993)*" in case someone wants to jump in._
```
```suggestion
_We use the "Introduced (version)" label to indicate the release in which a feature is introduced. But going backwards to identify everything is a big job. We received a suggestion to do this for field types, so I labeled it with "*[*_good first issue_*](https://github.com/opensearch-project/documentation-website/issues/6993)*" in case someone wants to jump in._
```

natebower commented on _posts/2024-06-03-so-you-want-better-opensearch-documentation.markdown:
```diff
@@ -0,0 +1,60 @@
+
+_We use the "Introduced (version)" label to indicate the release in which a feature is introduced. But going backwards to identify everything is a big job. We received a suggestion to do this for field types, so I labeled it with “*[*_good first issue_*](https://github.com/opensearch-project/documentation-website/issues/6993)*" in case someone wants to jump in._
+
+**Someone did make the comment that they did NOT believe that developers don’t like to write docs, especially in open source, but that we need to strive to continue to lower the barriers to contributing docs.** 
```
```suggestion
**Someone did make the comment that they do NOT believe that developers don't like to write docs, especially in open source, but that we need to strive to continue to lower the barriers to contributing docs.** 
```

natebower commented on _posts/2024-06-03-so-you-want-better-opensearch-documentation.markdown:
```diff
@@ -0,0 +1,60 @@
+
+_We are relying more and more on developers to write content and are looking at tools that make it easier and faster to write good documentation. We also have a great automated style checker built into our process that helps developers learn how to write better by correcting their mistakes._
+
+**One person asked what powered the site search, and the response that it is OpenSearch sparked a conversation about could we apply the new capabilities in OS like hybrid search and vectors to what powers the website. Kris Freedain spoke up with a “Who wants to collaborate on this?” and at least one person raised their hand ;-).**
```
```suggestion
**One person asked what powered the site search functionality, and the response that it is OpenSearch sparked a conversation about whether we could apply new OpenSearch capabilities, like hybrid search and vectors, to what powers the website. Kris Freedain spoke up with a "Who wants to collaborate on this?", and at least one person raised their hand ;-).**
```

natebower commented on _posts/2024-06-03-so-you-want-better-opensearch-documentation.markdown:
```diff
@@ -0,0 +1,60 @@
+
+_We are relying more and more on developers to write content and are looking at tools that make it easier and faster to write good documentation. We also have a great automated style checker built into our process that helps developers learn how to write better by correcting their mistakes._
+
+**One person asked what powered the site search, and the response that it is OpenSearch sparked a conversation about could we apply the new capabilities in OS like hybrid search and vectors to what powers the website. Kris Freedain spoke up with a “Who wants to collaborate on this?” and at least one person raised their hand ;-).**
```
Should we link Kris' name to his GitHub profile? It looks as though we link to his LinkedIn below, but is that the preferred method of getting in contact?

krisfreedain replied:
This is a great question. Since I do have a 'Community Member' profile on opensearch.org - it would make sense to link to it.  https://opensearch.org/community/members/kris-freedain.html 

It contains 4 links where the audience would be able to properly contact me. (LinkedIn, GitHub, X/Twitter, and Mastodon) 

I think this would be best for general guidance if this question comes up again. 
If the individual named in a future post does not have a Community Member profile, we could leave it up to them (linking to their LinkedIn, GitHub, X/Twitter, or Mastodon) - and possibly email.

natebower commented on _posts/2024-06-03-so-you-want-better-opensearch-documentation.markdown:
```diff
@@ -0,0 +1,60 @@
+
+_We are currently looking at how we can upgrade the search experience on the overall website. Please get in touch with [Kris Freedain](https://www.linkedin.com/in/krisfreedain/) if you are interested._
+
+_If you have more feedback on OpenSearch documentation, reach out on the [_forum_](https://forum.opensearch.org/c/feedback/6), [Slack](https://opensearch.slack.com/archives/C052GAV1MQF), or  [enter an issue](https://github.com/opensearch-project/documentation-website/issues/new/choose) in the documentation-website repo. We also welcome you to contribute your thoughts on the recently published [Op ...
```
```suggestion
_If you have more feedback on OpenSearch documentation, reach out on the [forum](https://forum.opensearch.org/c/feedback/6), [Slack](https://opensearch.slack.com/archives/C052GAV1MQF), or [enter an issue](https://github.com/opensearch-project/documentation-website/issues/new/choose) in the documentation-website repo. We also welcome you to contribute your thoughts on the recently published [OpenSearch Documentation Strategy](https://github.com/opensearch-project/documentation-website/issues/7189)._
```

natebower replied:
Should "documentation-website" be in code font?

natebower commented on _posts/2024-06-03-so-you-want-better-opensearch-documentation.markdown:
```diff
@@ -0,0 +1,60 @@
+
+_If you have more feedback on OpenSearch documentation, reach out on the [_forum_](https://forum.opensearch.org/c/feedback/6), [Slack](https://opensearch.slack.com/archives/C052GAV1MQF), or  [enter an issue](https://github.com/opensearch-project/documentation-website/issues/new/choose) in the documentation-website repo. We also welcome you to contribute your thoughts on the recently published [Op ...
+
+Thanks to the OpenSearch team for providing a great forum to connect with an incredible community. You can catch up on all the excitement in this [blog](https://opensearch.org/blog/OpenSearchCon-europe-2024-shines-in-berlin/), and I look forward to seeing everyone at [OpenSearchCon NA](https://opensearch.org/events/opensearchcon/2024/north-america/index.html)!
```
```suggestion
Thanks to the OpenSearch team for providing a great forum to connect with an incredible community. You can catch up on all the excitement in this [blog post](https://opensearch.org/blog/OpenSearchCon-europe-2024-shines-in-berlin/), and I look forward to seeing everyone at [OpenSearchCon North America](https://opensearch.org/events/opensearchcon/2024/north-america/index.html)!
```

natebower reviewed (commented):
@hdhalter Please see my comments and changes and let me know if you have any questions. 

@pajuric This is ready to go once Heather addresses my comments/changes and is happy with the draft.

hdhalter commented on _posts/2024-06-03-so-you-want-better-opensearch-documentation.markdown:
```diff
@@ -0,0 +1,60 @@
+ meta_description: <provided by marketing>
+---
+
+[Charlie Hull](https://opensourceconnections.com/team/charlie-hull/) and I, [Eric Pugh](https://opensourceconnections.com/team/eric-pugh/), OpenSource Connections, had the opportunity during the “Unconference” portion at [_OpenSearchCon EU_](https://opensearch.org/events/opensearchcon/2024/europe/index.html) to talk about the current state of the OpenSearch documentation and get some feedback fro ...
```
@epugh - any preference? I will change to"...from [OpenSource Connections](https://www.google.com/url?sa=t&source=web&rct=j&opi=89978449&url=https://opensourceconnections.com/&ved=2ahUKEwiqxrXG37WGAxUtM1kFHY5XDKgQFnoECAcQAQ&usg=AOvVaw0jRbdTI-7Nyxctwov1hOEu)." - with the link to your web page - unless you prefer something else.

epugh replied:
"from" works great!

hdhalter commented:
Thanks, @natebower ! 
@pajuric - I've addressed all the feedback.

pajuric commented:
@hdhalter -  I have this blog tentatively schedule for June 12. That's the first opening in the schedule due to 2.14 associated launch blogs.

hdhalter commented:
> @hdhalter - I have this blog tentatively schedule for June 12. That's the first opening in the schedule due to 2.14 associated launch blogs.

OK, sounds good. What keywords should I add?

hdhalter commented:
@pajuric - I added the keywords, thanks! I also changed the date to 6/18, so feel free to change based on the publish date. I didn't update the title, though.

pajuric commented:
@krisfreedain @nateynateynate - The blog is ready to publish today.

nateynateynate reviewed (approved):
Let's get it out there!
